headers/complex_h.md

```diff
@@ -1,8 +1,229 @@
 # \<complex.h\>
 
-Reference: [\<complex.h\>](https://en.cppreference.com/w/c/complex)
+C Reference: [\<complex.h\>][0]
 
-## Types
+Relevant Zig documentation:
+- [std.math.complex][20]
 
-## Functions
+## [C imaginary Type][1]
 
+This type allows you to write imaginary constants in C:
+
+```c
+double imaginary i = 2.0 * I; // pure imaginary
+double complex z = 1.0 + i; // complex number
+```
+
+In Zig, there are no imaginary types and you will be representing
+the same using [Complex](#zig-complex-type) struct:
+
+```zig
+const complex = @import("std").math.complex;
+
+const imaginary i = 2.0;
+const z = complex.Complex { re: 1.0, im: i };
+```
+
+!>  Note that imaginary types in C don't have Zig equivalent, you have to denote them as
+a [Complex](#zig-complex-type) type.
+
+
+## [Zig Complex Type][21]
+
+In Zig complex numbers are represented as `Complex` type.
```

Comment:
should note what the underlying number type is used in `Complex`

Reply:
Complex types take in numbers i guess as a generic. Didn't really check the source code. But sure, I will add it.